docs/measure/index.md

```diff
@@ -114,6 +114,236 @@ if __name__ == "__main__":
+   - **Prometheus**: Port `9090`
+
+4. **Optional Visualization Tools**  
+   Tools like Grafana can be integrated with Prometheus to create detailed visualizations of the metrics collected.
+
+---
+
+## Setup Guide
+
+### Step 1: Install and Start the Prometheus Push Gateway
+Choose either Option 1 (Binary) or Option 2 (Docker).
+
+#### Option 1: Download Binary
+1. Visit the [Prometheus Push Gateway Download Page](https://prometheus.io/download/#pushgateway).
+2. Download the appropriate binary for your operating system.
+3. Extract the binary:
+```sh
+   tar -xvzf prometheus-pushgateway*.tar.gz
+   cd prometheus-pushgateway-*
+```
+4. Start the Push Gateway:
+```sh
+./prometheus-pushgateway --web.listen-address=:9091
+```
+5. Verify the Push Gateway is running by visiting http://localhost:9091 in your browser.
+
+### Option 2: Using Docker
+1. Pull the official Prometheus Push Gateway Docker image:
+```sh
+docker pull prom/pushgateway
+```
+2. Run the Push Gateway in a container:
+```sh
+docker run -d -p 9091:9091 prom/pushgateway
+```
+3. Verify it is running by visiting http://localhost:9091 in your browser.
+
+### Step 2: Install and Configure Prometheus
+1. Visit the Prometheus [Prometheus Download Page](https://prometheus.io/download/#prometheus).
+2. Download the appropriate binary for your operating system.
+3. Extract the binary:
+```sh
+tar -xvzf prometheus*.tar.gz
+cd prometheus-*
+```
+4. Update the Prometheus configuration file `prometheus.yml` to scrape metrics from the Push Gateway:
+```sh
+scrape_configs:
+  - job_name: 'pushgateway'
+    honor_labels: true
+    static_configs:
+      - targets: ['localhost:9091']  # Replace with your Push Gateway URL
+```
+5. Start Prometheus:
+```sh
+./prometheus --config.file=prometheus.yml
+```
+6. Visit http://localhost:9090 in your browser, or use curl http://localhost:9090/api/v1/targets
+7. Verify Prometheus is running by visiting http://localhost:9090 in your browser.
+
+Zeus allows you to monitor energy and power consumption through different metrics, such as Histograms, Counters, and Gauges, which can be pushed to a Prometheus Push Gateway for further analysis. 
+
+---
+
+[`EnergyHistogram`][zeus.metric.EnergyHistogram] records energy consumption data for GPUs, CPUs, and DRAM in Prometheus Histograms. This is ideal for observing how often energy usage falls within specific ranges.
+
+You can customize the bucket ranges for each component (GPU, CPU, and DRAM), or let Zeus use default ranges.
+
+```python hl_lines="2 5-15"
+
+from zeus.metric import EnergyHistogram
+
+if __name__ == "__main__":
+    # Initialize EnergyHistogram
+    energy_histogram = EnergyHistogram(
+        cpu_indices=[0], 
+        gpu_indices=[0], 
+        prometheus_url='http://localhost:9091', 
+        job='training_energy_histogram'
+    )
+
+    for epoch in range(100):
+        # Start monitoring energy for the entire epoch
+        energy_histogram.begin_window(f"epoch_{epoch}_energy")
```

Comment:
This got me confused. Why is it okay for the window name to be different on each epoch? I thought the window name should always be the same (e.g., `training_epoch`), and each time when a window with exactly that name is measured (in this example snippet, every epoch), it will add a data point to the histogram metric uniquely identified by that window name.

Reply:
This is the largest concern of this review round. I see below that you also have:

```python
energy_histogram.begin_window(f"epoch_{epoch}_step_{step_idx}_energy")
```

And you're using the same instance of `energy_histogram`. I hope I'm wrong, but this has to mean that you're bunching together energy measurements from both training epoch and training step into the same `Histogram` metric with just different `window` labels, aren't you? If so, this defeats the whole purpose of histogram monitoring. The result of this example should be two completely independent `Histogram` metrics, one for monitoring training epochs (one datapoint added whenever `end_window("training_epoch")` is called) and another for monitoring training steps (one datapoint added whenever `end_window("training_step")` is called).